packages/app-tech-comm/src/Proposals/Propose.tsx

```diff
@@ -0,0 +1,98 @@
+// Copyright 2017-2019 @polkadot/ui-staking authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { I18nProps } from '@polkadot/react-components/types';
+import { TxSource, TxDef } from '@polkadot/react-hooks/types';
+import { Call } from '@polkadot/types/interfaces';
+
+import BN from 'bn.js';
+import React, { useEffect, useState } from 'react';
+import { registry } from '@polkadot/react-api';
+import { Extrinsic, InputNumber, TxModalNew as TxModal } from '@polkadot/react-components';
+import { useApi, useTx } from '@polkadot/react-hooks';
+import { createType } from '@polkadot/types';
+
+import translate from '../translate';
+
+interface Props extends I18nProps {
+  isOpen: boolean;
+  memberCount: number;
+  onClose: () => void;
+}
+
+function Propose ({ t, isOpen, onClose, memberCount = 0 }: Props): React.ReactElement<Props> {
+  const _hasThreshold = (threshold?: BN | null): boolean => {
+    return !!threshold && !threshold.isZero() && threshold.lten(memberCount);
+  };
+
+  const [method, setMethod] = useState<Call | null>(null);
+  const [[threshold, hasThreshold], setThreshold] = useState<[BN | null, boolean]>([
+    new BN(memberCount / 2 + 1),
+    true
+  ]);
+
+  const _onChangeThreshold = (threshold?: BN): void => {
+    setThreshold([threshold || null, _hasThreshold(threshold)]);
+  };
+
+  const _onChangeExtrinsic = (method?: Call): void => {
+    !!method && setMethod(method);
+  };
+
+  useEffect(
+    (): void => setThreshold([threshold, _hasThreshold(threshold)]),
+    [memberCount]
+  );
+
+  const { apiDefaultTxSudo } = useApi();
+  const txState = useTx(
```

Comment:
This whole things is just magic, no description of what is what. We will need to update this to be either an object of not having to pass `{...txState}` (more magic) to the Modal. Contrast this to TxButton (old), where it is obvious what you are passing.